### PR TITLE
Event.h, FunctionPointer.h, and mbed-sbrk are moving to mbed-util

### PR DESCRIPTION
**Pull request by rgrover**
refer to https://github.com/ARMmbed/core-util/pull/21 and https://github.com/ARMmbed/mbed-drivers/pull/85

@bogdanm @autopulated

### CODE REVIEW

File: source/minar.cpp [hunk 1]
```diff
@@ -16,7 +16,9 @@
  */
 
 #include "minar/minar.h"
+#ifndef YOTTA_CFG_POSIX
```
Comment (bogdanm) [inline source/minar.cpp]: This looks a bit ugly. Why do we still need `mbed.h` for?

Comment (rgrover) [inline source/minar.cpp]: I'll investigate why we need mbed.h

Comment (rgrover) [inline source/minar.cpp]: apparently the system builds in the absence of including mbed.h

**Comment by rgrover** [comment]
CR @bogdanm @autopulated

**Comment by autopulated** [comment]
That "mbed" targetDependency should probably be mbed-os, no?

**Comment by rgrover** [comment]
@autopulated we introduced mbed-os to distinguish it from mbed-classic. In using 'mbed' as a selector for my target-dependencies, I mean an mbed-hardware like platform.

**Comment by autopulated** [comment]
but mbed-drivers will only work on mbed-os platforms (and that seems very unlikely to change), so why be more generic than is necessary?

**Comment by rgrover** [comment]
mbed OS will run on Posix; so in my mind 'mbed-os' doesn't necessarily bring in mbed-drivers. We need a term to categorize platforms which require mbed-drivers. 'mbed-os' isn't it. I'm also not sure about 'mbed'. suggestions..

**Comment by bogdanm** [comment]
In the way we market mbed OS, I think `mbed-os` will do the trick. Our POSIX platform is just a mock platform for testing, not a full implementation of the OS. In the end though, I'm fine either way. I'm glad to see `#include "mbed.h"` removed from *minar.cpp*.

**Comment by hugovincent** [comment]
I wouldn't say the POSIX backend is "just a mock platform"; it's used for our testing, but also for our customers and users' testing and development, and can (when more developed) be a very useful and productive way of doing early development of mbed OS application logic.

File: module.json [hunk 2]
```diff
@@ -21,9 +21,13 @@
   "dependencies": {
     "compiler-polyfill": "~1.0.3",
     "minar-platform": "~0.3.0",
-    "mbed-drivers": "~0.6.0",
     "core-util": "~0.0.6"
   },
+  "targetDependencies": {
+    "mbed": {
+      "mbed-drivers": "~0.6.0"
```
Comment (0xc0170) [inline module.json]: Is this dep still needed? I hope this will be removed soon enough :)

**Comment by 0xc0170** [comment]
@rgrover Please rebase, there's conflict now with master.

**Comment by rgrover** [comment]
rebased. I've tested this against frdm-k64f-gcc.
@0xc0170 @bogdanm @autopulated

**Comment by 0xc0170** [comment]
LGTM

**Comment by rgrover** [comment]
@0xc0170 thanks for pointing out the un-necessary dependency. please review.

**Comment by 0xc0170** [comment]
+1

**Comment by rgrover** [comment]
bump

**Comment by bremoran** [comment]
+1

**Pull request merged**